src/components/Forms/Subscribe.js

```diff
@@ -16,98 +16,140 @@ const SubscribeSchema = Yup.object().shape({
-      initialValues={{
-        email_address: '',
-        first_name: '',
-      }}
-      validationSchema={SubscribeSchema}
-      onSubmit={values => {
-        // setTimeout(() => {
-        //   alert(JSON.stringify(values, null, 2))
-        // }, 500)
-        fetch('https://app.convertkit.com/forms/834199/subscriptions', {
+const PostSubmissionMessage = ({ response }) => {
+  console.log('Post submission response: ', response)
+
+  return <div>You have submitted with a status of {response.status}</div>
+}
+
+class SignUp extends React.Component {
+  state = {
+    submitted: false,
+  }
+
+  async handleSubmit(values) {
+    this.setState({ submitted: true, loading: true })
+    try {
+      const response = await fetch(
+        'https://app.convertkit.com/forms/834199/subscriptions',
+        {
           method: 'post',
           body: JSON.stringify(values, null, 2),
           headers: {
             Accept: 'application/json',
             'Content-Type': 'application/json',
           },
-        })
-      }}
-      render={({ errors, touched }) => (
-        <Form
-          css={css`
-            display: flex;
-            align-items: flex-end;
-            label:not(:first-of-type),
-            button {
-              margin-left: 10px;
-            }
-            .field-error {
-              display: block;
-              position: absolute;
-              color: ${theme.colors.red};
-              font-size: 80%;
-            }
-            input,
-            label {
-              width: 280px;
-            }
-            ${bpMaxSM} {
-              flex-direction: column;
-              align-items: center;
-              width: auto;
-              label {
-                margin: 15px 0 !important;
-              }
-              button {
-                margin-top: 20px;
-              }
-            }
-          `}
-        >
-          <label htmlFor="first_name">
-            First Name
-            <Field
-              aria-label="your first name"
-              aria-required="true"
-              name="first_name"
-              placeholder="Jane"
-              type="text"
-            />
-            <ErrorMessage
-              name="first_name"
-              component="span"
-              className="field-error"
-            />
-          </label>
-          <label htmlFor="email">
-            Email
-            <Field
-              aria-label="your email address"
-              aria-required="true"
-              name="email_address"
-              placeholder="jane@acme.com"
-              type="email"
-            />
-            <ErrorMessage
-              name="email_address"
-              component="span"
-              className="field-error"
+        },
+      )
+
+      const responseJson = await response.json()
+
+      this.setState({
+        submitted: true,
+        loading: false,
+        response: responseJson,
+        errorMessage: null,
+      })
+    } catch (error) {
```

Comment:
this can be tested by changing the url on line 34 to something that doesn't exist